packetbeat/sniffer/file.go

```diff
@@ -0,0 +1,106 @@
+package sniffer
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/tsg/gopacket"
+	"github.com/tsg/gopacket/layers"
+	"github.com/tsg/gopacket/pcap"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type fileHandler struct {
+	pcapHandle *pcap.Handle
+	file       string
+
+	loopCount, maxLoopCount int
+
+	topSpeed bool
+	lastTS   time.Time
+}
+
+func newFileHandler(file string, topSpeed bool, maxLoopCount int) (*fileHandler, error) {
+	h := &fileHandler{
+		file:         file,
+		topSpeed:     topSpeed,
+		maxLoopCount: maxLoopCount,
+	}
+	if err := h.open(); err != nil {
+		return nil, err
+	}
+
+	return h, nil
+}
+
+func (h *fileHandler) open() error {
+	tmp, err := pcap.OpenOffline(h.file)
+	if err != nil {
+		return err
+	}
+
+	h.pcapHandle = tmp
+	return nil
+}
+
+func (h *fileHandler) ReadPacketData() ([]byte, gopacket.CaptureInfo, error) {
+	data, ci, err := h.pcapHandle.ReadPacketData()
+	if err != nil {
+		if err != io.EOF {
+			return data, ci, err
+		}
+
+		h.pcapHandle.Close()
+		h.pcapHandle = nil
+
+		// TODO: Remove me.  still required to keep system tests working.
```

Comment:
waitstop is in seconds + I have had to add it to every test. The new publisher pipeline supports some wait stop behavior internally, but it's not used/exposed by any beat yet.

The `-waitstop` setting seems to be removed, in favor of a config file setting. I can try setting it to 300ms by default. But I didn't find this doesn't help much with test stability either. Ultimately we want #3404 (which should be possible with new publisher pipeline).